### ISSUE TITLE
Get rid of the empty taskbar space when taskbar overflow is shown

### ISSUE THREAD

**Issue by m417z** [ramensoftware/windhawk-mods#1131]
### Discussed in https://github.com/ramensoftware/windhawk-mods/discussions/1130

<div type='discussions-op-text'>

<sup>Originally posted by **s-h-a-d-o-w** October 23, 2024</sup>
(Also would be great to get rid of the "...")

![empty space](https://github.com/user-attachments/assets/136b40f7-b201-4d96-bf3c-34f2fdab594c)</div>

**Comment by s-h-a-d-o-w** [comment]
Just a suggestion but - might it make sense to include that as a toggle (or _toggles_ - if you implement both things) in the labels mod? At least in my mind, that's directly connected.

**Comment by m417z** [comment]
Yeah, maybe. On the other hand, perhaps there are users who aren't interested in customizing the labels, and just want to fix this bug. The taskbar overflow can be shown when labels aren't used, too. I'm not sure if the bug occurs in this case, though, that will require some investigation.

**Comment by s-h-a-d-o-w** [comment]
> The taskbar overflow can be shown when labels aren't used, too.

That particular mod supports that too though?
Anyway, I would be happy either way - just figured that it fits the theme and users might find it more easily. 🙂

**Comment by s-h-a-d-o-w** [comment]
Hm... while working on a theme today, I noticed that this for some reason stopped being a problem.

Maybe the combination of "minimum task bar item width: 20", "left and right padding size: 2" in the labels mod?

![space_reclaimed](https://github.com/user-attachments/assets/1b4d890d-18a7-43dc-aa0d-32ec8c97c602)

**Comment by m417z** [comment]
> minimum task bar item width: 20

That's why you aren't getting the overflow (the "..." button). That's what I meant in my comment here: https://github.com/ramensoftware/windhawk-mods/discussions/1130#discussioncomment-11032315.

If you'll open enough windows, it will eventually show up, and then the issue might be visible again.

**Comment by s-h-a-d-o-w** [comment]
Got it. I guess I just hadn't set it low enough. I think I had it at 40 or so before.
Personally, I just don't open THAT many windows since I want to be able to read at least ~4 characters of the window title.

Sidenote:
Would you accept a contribution for this Windows 10 Solid Dark theme?
I'm having a minor problem with it though - I can't get the running indicator to actually be full width. Maybe because of [this](https://github.com/ramensoftware/windhawk-mods/blob/b43bd5d00887601ed297b2366631494080f0dd3e/mods/taskbar-labels.wh.cpp#L1047)?

**Comment by m417z** [comment]
> Would you accept a contribution for this Windows 10 Solid Dark theme?

Which customizations does it have except for the solid background? If it's only that, I think that having it as a theme is a bit of an overkill.

> I can't get the running indicator to actually be full width. Maybe because of [this](https://github.com/ramensoftware/windhawk-mods/blob/b43bd5d00887601ed297b2366631494080f0dd3e/mods/taskbar-labels.wh.cpp#L1047)? (The gap on the right IS about 6 pixels)

Having the running indicators full width can be messy, depending on the taskbar configuration, and Taskbar Labels implements some workarounds to make it work it all cases. Why implement it manually and not use the Taskbar Labels mod for the full width indicators?

**Comment by s-h-a-d-o-w** [comment]
Probably best to just show you.

> Why implement it manually and not use the Taskbar Labels mod for the full width indicators?

I don't, as you'll see. It's a combination of using the labels mod with the following:

```
  "controlStyles[1].target": "Taskbar.TaskListButton",
  "controlStyles[1].styles[0]": "CornerRadius=0",
  "controlStyles[2].target": "Taskbar.TaskListLabeledButtonPanel@CommonStates > Border#BackgroundElement",
  "controlStyles[2].styles[0]": "Background@ActiveNormal=#FF777777",
  "controlStyles[2].styles[1]": "Background@InactivePointerOver=#FF444444",
  "controlStyles[2].styles[2]": "Background@ActivePointerOver=#FF777777",
  "controlStyles[2].styles[3]": "BorderThickness=0",
  "controlStyles[2].styles[4]": "Margin=0,0,0,0",
  "controlStyles[3].target": "Taskbar.TaskListLabeledButtonPanel@RunningIndicatorStates > Rectangle#RunningIndicator",
  "controlStyles[3].styles[0]": "Height=2",
  "controlStyles[4].target": "Taskbar.TaskListLabeledButtonPanel",
  "controlStyles[4].styles[0]": "Padding=1,0,0,0",
  "controlStyles[5].target": "Rectangle#RunningIndicator",
  "controlStyles[5].styles[0]": "RadiusX=0",
  "controlStyles[5].styles[1]": "RadiusY=0",
  "controlStyles[5].styles[2]": "Margin=0,0,0,0",
  "controlStyles[6].target": "Rectangle#BackgroundStroke",
  "controlStyles[6].styles[0]": "Height=0",
  "controlStyles[7].target": "Rectangle#BackgroundFill",
  "controlStyles[7].styles[0]": "Fill=#FF1C1C1C",
  "controlStyles[8].target": "Taskbar.TaskListButtonPanel#ExperienceToggleButtonRootPanel",
  "controlStyles[8].styles[0]": "Padding=0,0,0,0",
```
